pkg/network/common/egressip.go

```diff
@@ -211,8 +211,9 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *osdnv1.HostSubnet) {
 		_, cidr, err := net.ParseCIDR(hs.Subnet)
 		if err != nil {
 			klog.Errorf("Could not parse HostSubnet %q CIDR: %v", hs.Name, err)
+		} else {
+			sdnIP = GenerateDefaultGateway(cidr).String()
 		}
-		sdnIP = GenerateDefaultGateway(cidr).String()
```

Comment:
We should ignore the HostSubnet completely if it is invalid. Rather, we should never even get to this point if the HostSubnet has any invalid data.

We call `ValidateHostSubnetEgress` in `handleAddOrUpdateHostSubnet`/`handleDeleteHostSubnet`, but that function doesn't check `hs.Subnet`, which is a bug. `ValidateHostSubnetEgress` should just call `ValidateHostSubnet` as well, since we require both `hs.Subnet` and `hs.HostIP` to be valid in the egress-handling code.

Then we can safely assume that `net.ParseCIDR(hs.Subnet)` doesn't return an error (and thus we can assume that we'll always have an `sdnIP` for every node).

Reply:
After `ValidateHostSubnet` is added to `ValidateHostSubnetEgress` what should we do here? return from function or ignore the error? I would prefer the former